scp/filesystem.go

```diff
@@ -10,20 +10,37 @@ import (
 	"time"
 
 	"github.com/charmbracelet/ssh"
+	"github.com/pkg/sftp"
 )
 
-type fileSystemHandler struct{ root string }
+// FileSystemHandler is Handler and SFTP implementation for a given root path.
+type FileSystemHandler struct{ root string }
 
-var _ Handler = &fileSystemHandler{}
+var (
+	_ Handler = &FileSystemHandler{}
+	_ Sftp    = &FileSystemHandler{}
+)
 
 // NewFileSystemHandler return a Handler based on the given dir.
-func NewFileSystemHandler(root string) Handler {
-	return &fileSystemHandler{
+func NewFileSystemHandler(root string) *FileSystemHandler {
+	return &FileSystemHandler{
 		root: filepath.Clean(root),
 	}
 }
 
-func (h *fileSystemHandler) chtimes(path string, mtime, atime int64) error {
+func (h *FileSystemHandler) ServerOptions() []sftp.ServerOption {
+	return []sftp.ServerOption{
+		func(s *sftp.Server) error {
+			abs, err := filepath.Abs(h.root)
+			if err != nil {
+				return err
+			}
+			return sftp.WithServerWorkingDirectory(abs)(s)
```

Comment:
If this is the only option, maybe it's not worth it introducing a new interface `Sftp` under `scp`. This might be confusing since scp != sftp and they're totally different protocols.

Instead, the filesystem scp handler could have a `WorkingDirectory()` or a `RootDirectory()`

Reply:
yeah, you're probably right... I'm actually almost thinking we can provide just the `WithSubsystem` and not touch any of this, maybe add it as an example only 🤔